mlflow/gensim.py

```diff
@@ -0,0 +1,207 @@
+    """
+    import gensim
+
+    path = os.path.abspath(path)
+    if os.path.exists(path):
+        raise MlflowException("Path '{}' already exists".format(path))
+
+    model_data_subpath = "model.pkl"
+    model_data_path = os.path.join(path, model_data_subpath)
+    os.makedirs(path)
+
+    # Save an Gensim model
+    _save_model(gen_model=gen_model, output_path=model_data_path)
+
+    conda_env_subpath = "conda.yaml"
+    if conda_env is None:
+        conda_env = get_default_conda_env()
+    elif not isinstance(conda_env, dict):
+        with open(conda_env, "r") as f:
+            conda_env = yaml.safe_load(f)
+    with open(os.path.join(path, conda_env_subpath), "w") as f:
+        yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
+
+    pyfunc.add_to_model(mlflow_model, loader_module="mlflow.gensim",
+                        data=model_data_subpath, env=conda_env_subpath)
+
+    mlflow_model.add_flavor(FLAVOR_NAME, genism_version=gensim.__version__, data=model_data_subpath)
+    mlflow_model.save(os.path.join(path, "MLmodel"))
+
+
+def log_model(gen_model, artifact_path, conda_env=None, registered_model_name=None):
+    """
+    Log an Gensim model as an MLflow artifact for the current run.
+
+    :param gen_model: gensim model to be saved.
+    :param artifact_path: Run-relative artifact path.
+    :param conda_env: Either a dictionary representation of a Conda environment or the path to a
+                      Conda environment yaml file. If provided, this describes the environment
+                      this model should be run in. At minimum, it should specify the dependencies
+                      contained in :func:`get_default_conda_env()`. If ``None``, the default
+                      :func:`get_default_conda_env()` environment is added to the model.
+                      The following is an *example* dictionary representation of a Conda
+                      environment::
+
+                        {
+                            'name': 'mlflow-env',
+                            'channels': ['defaults'],
+                            'dependencies': [
+                                'python=3.7.0',
+                                'pip': [
+                                    'gensim==3.8.3'
+                                ]
+                            ]
+                        }
+    :param registered_model_name: (Experimental) If given, create a model version under
+                                  ``registered_model_name``, also creating a registered model if one
+                                  with the given name does not exist.
+
+
+                      .. code-block:: python
+
+                        from mlflow.models.signature import infer_signature
+                        train = df.drop_column("target_label")
+                        signature = infer_signature(train, model.predict(train))
+
+    """
+    Model.log(artifact_path=artifact_path,
```

Comment:
Can we add pass-through parameters for `signature` and `input_example` similar to other flavors?

Reply:
Sorry for very late reply. I have had them included, but they were failing I don't know why

Reply:
As you can see in the comment above, I don't really understand why it fails...